### PR TITLE
Message encryption

### PR DESCRIPTION
**Pull request by jborean93**
Minor changes around how the library encrypts and decrypts messages so it can handle larger messages. Has also more easily exposed the TLS wrapping and unwrapping methods.

### CODE REVIEW

**Comment by jborean93** [comment]
Coverage results on Appveyor

image

File: requests_credssp/credssp.py [hunk 1]
```diff
@@ -309,15 +310,55 @@ def _send_encrypted_credentials(self, response, **kwargs):
...
+        server.
+
+        :param data: a byte string of data to encrypt
+        :return: a byte string of the encrypted data
+        """
+        self.tls_connection.send(data)
+        encrypted_data = b''
+
+        while True:
+            try:
```
Comment (nitzmahone) [inline requests_credssp/credssp.py]: Should probably have a sanity counter w/ forced exit on both of these loops around some multiple of len(data)- if the TLS session is borked without throwing (seen it happen, sadly), we'd technically go forever on this.

File: requests_credssp/credssp.py [hunk 2]
```diff
@@ -309,15 +310,55 @@ def _send_encrypted_credentials(self, response, **kwargs):
...
+        between the client and the server.
+
+        :param encrypted_data: the byte string of the encrypted data
+        :return: a byte string of the decrypted data
+        """
+        self.tls_connection.bio_write(encrypted_data)
+        data = b''
+
+        while True:
+            try:
```
Comment (nitzmahone) [inline requests_credssp/credssp.py]: sanity counter as above

File: README.md [hunk 3]
```diff
@@ -67,10 +83,43 @@ r = requests.get("https://server:5986/wsman", auth=credssp_auth)
 ...
 ```
 
-Backlog
--------
+#### Message Encryption
+
+You can use this library to encrypt and decrypt messages sent ot and from the
```
Comment (nitzmahone) [inline README.md]: s/ot/to/

**Comment by jborean93** [comment]
@nitzmahone changes have been made, let me know if you think it could be done differently.

**Comment by jborean93** [comment]
Hey @nitzmahone I've updated the PR and added some logic to calculate the trailer length here https://github.com/jborean93/requests-credssp/blob/f6d435a500439009a1bf26fca1d050c429149e28/tests/functional/test_credssp_connection.py#L113-L156. Unfortunately I couldn't find an easy way to do any of this stuff and is reliant on parsing the cipher name returned by OpenSSL. There are also various scenarios at play here but I have tested the following so far

* AES-GCM cipher which has a fixed trailer length
* TLS 1.2 with AES-CBC on different MAC hashes
* TLS 1.0 with AES-CBC on different MAC hashes

I need to do some more testing around this but any ideas on how to make this less hacky I am all ears.

**Pull request merged**